tests/utils/syscall_handler.py

```diff
@@ -262,6 +264,25 @@ def storage_write(self, segments, syscall_ptr):
             value=segments.memory[syscall_ptr + 2],
         )
 
+    def replace_class(self, segments, syscall_ptr):
+        """
+        Record the replaced class hash in the internal mock object and update the class_hash attribute.
+
+        Syscall structure is:
+
+            struct ReplaceClass {
+                selector: felt,
+                class_hash: felt,
+            }
+        """
+        class_hash = segments.memory[syscall_ptr + 1]
+        address = self.contract_address
+        self.mock_replace_class(
+            address=address,
```

Comment:
I am not sure that we want to add the address here, as the mock was meant to record the actual syscall, and address is not part of it. I don't see the value added by this change

Reply:
This way we can track the replace classes applied to specific accounts?

Reply:
but the `self.contract_address` is not updated in `call_contract`, so there is no notion of address

Reply:
ok, yes I misunderstood this